test/unit/test_matrix.cpp

```diff
@@ -0,0 +1,365 @@
+               << "expected " << el2({i, j}) << " == " << *ptr({i, j}) << std::endl;
+      }
+    }
+  }
+}
+
+template <class T, Device device>
+void checkFromExisting(T* p, const matrix::LayoutInfo& layout, Matrix<const T, device>& matrix) {
+  auto el = [](const GlobalElementIndex& index) {
+    SizeType i = index.row();
+    SizeType j = index.col();
+    return TypeUtilities<T>::element(i + 0.001 * j, j - 0.01 * i);
+  };
+  auto ptr = [p, layout](const GlobalElementIndex& index) { return getPtr(p, layout, index); };
+  const auto& size = layout.size();
+
+  for (SizeType j = 0; j < size.cols(); ++j) {
+    for (SizeType i = 0; i < size.rows(); ++i) {
+      *ptr({i, j}) = el({i, j});
+    }
+  }
+
+  EXPECT_EQ(MatrixBase(layout), matrix);
+  CHECK_MATRIX_PTR(ptr, matrix);
+  CHECK_MATRIX_EQ(el, matrix);
+}
+
+#define CHECK_FROM_EXISTING(p, layout, mat) \
+  do {                                      \
+    SCOPED_TRACE("");                       \
+    checkFromExisting(p, layout, mat);      \
+  } while (0)
+
+std::vector<std::tuple<GlobalElementSize, TileElementSize, SizeType, std::size_t, std::size_t>> values(
+    {{{31, 17}, {7, 10}, 31, 7, 341},     // Scalapack like layout
+     {{31, 17}, {7, 11}, 33, 7, 363},     // with padding (ld)
+     {{31, 17}, {7, 11}, 47, 11, 517},    // with padding (row)
+     {{31, 17}, {7, 11}, 31, 7, 348},     // with padding (col)
+     {{29, 41}, {13, 11}, 13, 143, 429},  // Tile like layout
+     {{29, 41}, {13, 11}, 17, 183, 549},  // with padding (ld)
+     {{29, 41}, {13, 11}, 13, 146, 438},  // with padding (row)
+     {{29, 41}, {13, 11}, 13, 143, 436},  // with padding (col)
+     {{29, 41}, {13, 11}, 13, 143, 419},  // compressed col_offset
+     {{0, 0}, {1, 1}, 1, 1, 1}});
+
+TYPED_TEST(MatrixTest, ConstructorExisting) {
+  using Type = TypeParam;
+
+  for (const auto& v : values) {
+    auto size = std::get<0>(v);
+    auto block_size = std::get<1>(v);
+    auto ld = std::get<2>(v);
+    auto row_offset = std::get<3>(v);
+    auto col_offset = std::get<4>(v);
+
+    matrix::LayoutInfo layout(size, block_size, ld, row_offset, col_offset);
+    memory::MemoryView<Type, Device::CPU> mem(layout.minMemSize());
+
+    Matrix<Type, Device::CPU> mat(layout, mem(), mem.size());
+
+    CHECK_FROM_EXISTING(mem(), layout, mat);
+  }
+}
+
+TYPED_TEST(MatrixTest, ConstructorExistingConst) {
+  using Type = TypeParam;
+
+  for (const auto& v : values) {
+    auto size = std::get<0>(v);
+    auto block_size = std::get<1>(v);
+    auto ld = std::get<2>(v);
+    auto row_offset = std::get<3>(v);
+    auto col_offset = std::get<4>(v);
+
+    matrix::LayoutInfo layout(size, block_size, ld, row_offset, col_offset);
+    memory::MemoryView<Type, Device::CPU> mem(layout.minMemSize());
+
+    const Type* p = mem();
+    Matrix<const Type, Device::CPU> mat(layout, p, mem.size());
+
+    CHECK_FROM_EXISTING(mem(), layout, mat);
+  }
+}
+
+/// @brief Returns true if the first @p ready futures are ready.
```

Comment:
I would say _"Return turn if JUST the firsts [...]"_ and try to find a more explicit name for the function.